### ISSUE TITLE
Libxml2:x64-windows-static cannot be used by cmake projects that find libxml2 twice in different subdirectory

### ISSUE THREAD

**Issue by gnaggnoyil** [microsoft/vcpkg#19713]
**Describe the bug**
If a cmake project have `find_package(LibXml2)` both in top directory `CMakeLists.txt` and in one of its subdirectory `CMakeLists.txt`, usage of vcpkg port "libxml2" will fail during project configuration time.

**Environment**
- OS: OS: Windows 10 20H1 19043
- Compiler: Compiler: MSVC 19.29.30133 installed with VS2019
- CMake version: 3.21.1

**To Reproduce**
1. `.\vcpkg.exe install "libxml2[*]"` --triplet=x64-windows-static
2. Create a project directory with its contents described below.
3. In project directory, run `cmake -DCMAKE_TOOLCHAIN_FILE=<...>\vcpkg\scripts\buildsystems\vcpkg.cmake" -DVCPKG_TARGET_TRIPLET=x64-windows-static .`

**Expected behavior**
The project should be configured and generated successfully

**Actual behavior**
Console output was like:
```
-- Found LibXml2: optimized;<...>/vcpkg/installed/x64-windows-static/lib/libxml2.lib;debug;<...>/vcpkg/installed/x64-windows-static/debug/lib/libxml2.lib (found version "2.9.12")
CMake Error at <...>/vcpkg/installed/x64-windows-static/share/libxml2/vcpkg-cmake-wrapper.cmake:47 (target_link_libraries):
  Cannot specify link libraries for target "LibXml2::LibXml2" which is not
  built by this project.
Call Stack (most recent call first):
  <...>/vcpkg/scripts/buildsystems/vcpkg.cmake:738 (include)
  dir1/CMakeLists.txt:6 (find_package)


-- Configuring incomplete, errors occurred!
See also "C:/test_proj1/CMakeFiles/CMakeOutput.log".
See also "C:/test_proj1/CMakeFiles/CMakeError.log".
```

**Project content**

* `.\CMakeLists.txt`
    ```cmake
    cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
    project("test1")
    find_package(LibXml2 REQUIRED)
    add_executable(main main.cpp)
    target_link_libraries(main PUBLIC LibXml2::LibXml2)
    add_subdirectory(dir1)
    ```
* `.\main.cpp`
    ```c++
    int main(){}
    ```
* `.\dir1\CMakeLists.txt`
    ```cmake
    cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
    project("dir1")
    find_package(LibXml2 REQUIRED)
    add_executable(dir1 dir1.cpp)
    target_link_libraries(dir1 PUBLIC LibXml2::LibXml2)
    ```
* `.\dir1\dir1.cpp`
    ```c++
    int main(){}
    ```

**Comment by gnaggnoyil** [comment]
This issue seems to be related to this commit: https://github.com/microsoft/vcpkg/pull/17945/commits/f8558bb40ecbfb64e54652681169a2da5b07fd62
Anyone have any ideas about it?

**Comment by dg0yt** [comment]
See PR #19120.

**Comment by JackBoosY** [comment]
So I think this issue was fixed by #19120.
